### ISSUE TITLE
Not able to load the BertForSequenceClassification model from huggingface

### ISSUE THREAD

**Issue by rkoystart** [triton-inference-server/server#3548]
**Description**
Hi, i am not able to load the BertForSequenceClassification model from huggingface in the triton inference server

**Triton Information**
v21.09

Are you using the Triton container or did you build it yourself?
I am using the triton container nvcr.io/nvidia/tritonserver:21.09-py3

**To Reproduce**
I used the following script to convert the bert model to traced model. And i wanted to load this model in the triton inference server.
```
import torch
from transformers import BertForSequenceClassification, BertTokenizer

input_ids = torch.tensor([[2,3,4,5]]).long()
mask = input_ids != 1
mask = mask.long()

class MyModel(torch.nn.Module):
    def __init__(self):
       super().__init__()
       self.model =  BertForSequenceClassification.from_pretrained("bert-base-uncased")

    def forward(self,data, attention_mask):
      out2 = self.model(data, attention_mask=attention_mask) 
      return out2[0]



pt_model = MyModel().eval()
print(pt_model(input_ids, mask))
traced_model = 	torch.jit.trace(pt_model,(input_ids,mask))
traced_model.save("model.pt")

```
the config.pbtxt file is 
```
name: "bert"
platform: "pytorch_libtorch"
input [
 {
    name: "input__0"
    data_type: TYPE_INT32
    dims: [1, 512]
  },
  input [
 {
    name: "input__1"
    data_type: TYPE_INT32
    dims: [1, 512]
  }
]
output {
    name: "output__0"
    data_type: TYPE_FP32
    dims: [1, 2]
  }

```
And now runned the command
`sudo docker run --rm -p8000:8000 -p8001:8001 -p8002:8002 -v /home/rkoy/server/model_repository:/models nvcr.io/nvidia/tritonserver:21.09-py3 tritonserver --model-repository=/models
`
```
I got this error
UNAVAILABLE: Internal: failed to load model 'bert':                                                                       |                                                             
|       |         | Arguments for call are not valid.                                                                                        |                                                             
|       |         | The following variants are available:                                                                                    |                                                             
|       |         |                                                                                                                          |                                                             
|       |         |   aten::gelu(Tensor self, bool approximate) -> (Tensor):                                                                 |                                                             
|       |         |   Argument approximate not provided.                                                                                     |                                                             
|       |         |                                                                                                                          |                                                             
|       |         |   aten::gelu.out(Tensor self, bool approximate, *, Tensor(a!) out) -> (Tensor(a!)):                                      |                                                             
|       |         |   Argument approximate not provided.                                                                                     |                                                             
|       |         |                                                                                                                          |                                                             
|       |         | The original call is:                                                                                                    |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/functional.py(1313): gelu                                 |                                                             
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py(425): forward            |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(534): _slow_forward                     |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(548): __call__                          |                                                             
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py(523): feed_forward_chunk |                                                             
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/transformers/modeling_utils.py(2349): apply_chunking_to_forward    |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py(511): forward            |                                                             
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(534): _slow_forward                     |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(548): __call__                          |                                                             
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py(583): forward            |                                                             
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(534): _slow_forward                     |                                                             
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(548): __call__                          |                                                             
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py(996): forward            |                                                             |       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(534): _slow_forward                     |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(548): __call__                          |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py(1530): forward           |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(534): _slow_forward                     |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(548): __call__                          |
|       |         | convert_pytorch_model_to_jit.py(17): forward                                                                             |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(534): _slow_forward                     |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py(548): __call__                          |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/jit/__init__.py(1027): trace_module                          |
|       |         | /data/rkoy/anaconda3/lib/python3.8/site-packages/torch/jit/__init__.py(873): trace                                  |
|       |         | convert_pytorch_model_to_jit.py(25): <module>                                                                            |
|       |         | Serialized   File "code/__torch__/transformers/models/bert/modeling_bert.py", line 178                                   |
|       |         |   def forward(self: __torch__.transformers.models.bert.modeling_bert.BertIntermediate,                                   |
|       |         |     argument_1: Tensor) -> Tensor:                                                                                       |
|       |         |     input = torch.gelu((self.dense).forward(argument_1, ))                                                               |
|       |         |             ~~~~~~~~~~ <--- HERE                                                                                         |
|       |         |     return input                                                                                                         |
|       |         | class BertOutput(Module):                          

```
**Expected behavior**
I expected that the bert model would have been loaded successfully without any error

Would like to know to mistake i am making and also a solution for it.

**Comment by CoderHam** [comment]
@rkoystart this looks to be a bug with tracing. 
Have you tried to run the torchscript model outside Triton successfully?
What version of PyTorch did you use to produce the model? 
Have you tried using scripting to produce the torchscript model?

**Comment by zirui** [comment]
i am having the same issue  @CoderHam 
1. Have you tried to run the torchscript model outside Triton successfully?
yes
2. What version of PyTorch did you use to produce the model?
'pytorch 1.8.1+cu101' and ''1.10.0+cu111' both failed

3. Have you tried using scripting to produce the torchscript model?

 code snippets to convert to torchscript model
```python
traced_model = torch.jit.trace(model, dummy_input)
torch.jit.save(traced_model, torch_script_model_path)
```

i test the model in two `tis` dockers:

| docker  |successed? |
|---|---|
|  nvcr.io/nvidia/tritonserver:20.11-py3 |  yes |
|  nvcr.io/nvidia/tritonserver:21.10-py3 | no  |

**Comment by EvGe22** [comment]
Hi. Having the same problem running `nvcr.io/nvidia/tritonserver:21.11-py3` and `transformers==2.10` (I know it's kinda old)

Have managed to _"solve"_ this by removing the `if` in the transformers code [here](https://github.com/huggingface/transformers/blob/v2.10.0/src/transformers/activations.py#L32-L35) and just hard coding the `gelu = _gelu_python`

This allowed me to run the model. It even seems to run fine. There are some small differences between the predictions of the PyTorch model and the the TritonServer one, but they are negligible in my case. The speed seems almost the same. In my case the Triton server is even sometimes slower.

But I have only been testing with the `dynamic batching` option turned off and I already had some internal code that turns text into torch.Tensor. So, in order to use the Triton model I have to convert every Tensor into numpy array first. + the GRPC delay.

In my case, a PyTorch version takes ~1.61 seconds to run 1973 items (batches of size 100), Triton takes ~1.64 seconds.
However, when we split the data into randomly sized (1-100) batches the times differ more: 2.384 PyTorch vs 2.75 Triron
The random seed used to split the batches was fixed, and I have run 10 experiments, these are the mean values. The std is quite low - 0.02 seconds in both cases.

**Comment by CoderHam** [comment]
> However, when we split the data into randomly sized (1-100) batches the times differ more: 2.384 PyTorch vs 2.75 Triron

@EvGe22 The time can be due to overheads like network overhead. Can you use perf_analyzer to  give us a breakdown of the same?

**Comment by CoderHam** [comment]
@zirui did you use the pytorch NGC containers in both cases?

> What version of PyTorch did you use to produce the model?
'pytorch 1.8.1+cu101' and ''1.10.0+cu111' both failed

Since it worked in 20.11 I am certain this is due to a version issue or internal to pytorch.

**Comment by zirui** [comment]
> 

no, i use pytorch downloaded from the [offical pytorch sites](https://download.pytorch.org/whl/cu111/torch_stable.html)

**Comment by zirui** [comment]
> @zirui did you use the pytorch NGC containers in both cases?
> 
> > What version of PyTorch did you use to produce the model?
> > 'pytorch 1.8.1+cu101' and ''1.10.0+cu111' both failed
> 
> Since it worked in 20.11 I am certain this is due to a version issue or internal to pytorch.

no, i use pytorch downloaded from the [offical pytorch sites](https://download.pytorch.org/whl/cu111/torch_stable.html)

**Comment by CoderHam** [comment]
@rkoystart the reason you see this issue in 21.10 which is based on the PyTorch version [1.10.0a0+0aef44c](https://github.com/pytorch/pytorch/commit/0aef44c) but not official PyTorch 1.10.0 is because the NGC PyTorch version 1.10.0a0+0aef44c is not the same as 1.10.0.

The way our release works is that we branch off a short time before 1.10.0 is finalized and sometimes this causes undesired effects. Please try the same with the 21.11 release.

**Comment by naor2013** [comment]
@CoderHam It seems like the issue is because Nvidia's PyTorch is using the unmerged https://github.com/pytorch/pytorch/pull/61439 pull request. This means that if you train your model (or at least covert your model to TorchScript) using "regular" (non-nvidia) PyTorch and then try to use it in Triton, it does not work.

Is there any way that Nvidia will either release a PyTorch version which is not incompatible with regular PyTorch or if the person who is working on that pull request is from Nvidia, maybe they can make sure this pull request isn't breaking "regular" PyTorch?

**Comment by naor2013** [comment]
@CoderHam Can you let me know if this is something that someone in Nvidia is investigating? This bug doesn't allow us to deploy models as frequently as we're used to because it makes our process much more manual.

**Comment by dzier** [comment]
@naor2013  we just released a new [version of Triton](https://github.com/triton-inference-server/server/releases/tag/v2.17.0) which contains Pytorch version [1.11.0a0+b6df043](https://github.com/pytorch/pytorch/commit/b6df043).

**Comment by naor2013** [comment]
> @naor2013 we just released a new [version of Triton](https://github.com/triton-inference-server/server/releases/tag/v2.17.0) which contains Pytorch version [1.11.0a0+b6df043](https://github.com/pytorch/pytorch/commit/b6df043). Can you please try the new version to see if the problem persists?

@dzier I didn't try it yet in Triton but I opened the image and looked in Pytorch and it seemed that the relevant uncompatible code remained the same (aka uncompatible).
I will test with Triton itself but I don't think that will change the result.

**Comment by tuxedocat** [comment]
Try exporting with `torch.jit.trace(model)` on NGC-PyTorch 21.12 `nvcr.io/nvidia/pytorch:21.12-py3` image.

I also tried exporting DistilBERT in huggingface's impl. (`transofmers==4.15.0`) and running it on NGC Triton 21.12 with libtorch (`21.12-pyt-python-py3`). Using NGC-PyTorch-21.12 image to trace the model saved me.

First failed attempt was to use stable `torch==1.10.1` when exporting but it resulted as failed due to GELU function signature issue reported above. Since NGC Triton + LibTorch image has libtorch corresponding to 1.11.0a0+b6df043, the torchscript model itself should be traced using the same version of pytorch.

**Comment by dyastremsky** [comment]
Closing issue due to lack of activity. Please re-open the issue if you would like to follow up with this issue.

**Comment by shaked571** [comment]
I am also suffering the same issue, already spend two days trying to figure it out.

Did any of you guys managed to solve it? @naor2013 @zirui ?
